cogboard-webapp/src/components/UserLogin.js

```diff
@@ -1,20 +1,31 @@
-import React from 'react';
+import React, {useEffect} from 'react';
 import { useDispatch, useSelector } from 'react-redux';
+import { Button, IconButton, TextField, Typography } from '@material-ui/core';
+import { AccountCircle, PowerSettingsNew } from '@material-ui/icons';
 
-import { useDialogToggle, useFormData } from '../hooks';
+import { useFormData, useToggle } from '../hooks';
 import { login, logout } from '../actions/thunks';
-
-import { TextField, Button, IconButton, Typography } from '@material-ui/core';
-import { AccountCircle, PowerSettingsNew } from '@material-ui/icons';
 import { StyledFieldset } from './styled';
 import AppDialog from './AppDialog';
+import SnackbarWithVariant from "./SnackbarWithVariant";
 
 const UserLogin = () => {
   const dispatch = useDispatch();
-  const { values, handleChange } = useFormData({ username: '', password: ''});
-  const [dialogOpened, openDialog, handleDialogClose] = useDialogToggle();
+  const {values, handleChange} = useFormData({username: '', password: ''});
   const errorMsg = useSelector(({app}) => app.loginErrorMessage);
   const jwToken = useSelector(({app}) => app.jwToken);
+  const isUserLogged = !!jwToken;
+  const [dialogOpened, openDialog, handleDialogClose] = useToggle();
+  const [loginSnackbarOpened, openLoginSnackbar, handleLoginSnackbarClose] = useToggle();
+  const [logoutSnackbarOpened, openLogoutSnackbar, handleLogoutSnackbarClose] = useToggle();
+  const onLoginButtonClick = () => {
+    if(isUserLogged) {
+      handleDialogClose();
+      openLoginSnackbar();
+    }
+  };
+
+  useEffect(onLoginButtonClick, [isUserLogged]);
```

Comment:
Grat idea of using this hook, but onLoginButtonClick name miss what really happens here. I suggest moving whole function to use effect call. It is considered a good practice therefore linter will warn you when you forget to put any dependency into array. It is also clear that effects are invoked on every render, but based on dependency change.